lib/phoenix_live_view/channel.ex

```diff
@@ -613,6 +615,19 @@ defmodule Phoenix.LiveView.Channel do
     {:stop, :shutdown, :no_session}
   end
 
+  defp check_version_match(params) do
```

Comment:
You can collect the version at compile time with this?

    @version Mix.Project.config()[:version]

My suggestion though is to match only on the prefix, so I would do something like this:

    @version Mix.Project.config()[:version]
    @prefix_version String.split(@version, ".") |> Enum.drop(-1) |> Enum.join(".") |> Kernel.<>(".")

And then inside you do:

    if should_display_warning? and is_string(client_version) and not String.starts_with?(client_version, @prefix_version) do

That's because 0.14.2 works with 0.14.1 but there is no guarantee it works with 0.15 or 0.13. WDYT?

Reply:
Hmm, on one hand I think that makes sense since it is always guaranteed to work and not break, on the other hand, i think this feature is also really useful to track why something might not be working, ie: with the `push_event` changes, maybe the dev using it updates the mix part but forgets the npm part, so calling `push_event` won't break, but the js client won't get the call because it has not been updated yet. Keeping it specific I think would help with this type of problems when one might spend some time debugging only to find out it was a mismatch error.